module/state_synchronization/indexer/indexer_core.go

```diff
@@ -150,10 +144,11 @@ func (c *IndexerCore) indexEvents(blockID flow.Identifier, events flow.EventsLis
 	return err
 }
 
-func (c *IndexerCore) indexRegisters(payloads []*ledger.Payload, height uint64) error {
-	regEntries := make(flow.RegisterEntries, len(payloads))
+func (c *IndexerCore) indexRegisters(registers map[ledger.Path]*ledger.Payload, height uint64) error {
+	regEntries := make(flow.RegisterEntries, len(registers))
```

Comment:
you can drop the `j` and just append
```suggestion
regEntries := make(flow.RegisterEntries, 0, len(registers))
```